swap/protocol.go

```diff
@@ -64,9 +71,28 @@ func (s *Swap) Stop() error {
 	return nil
 }
 
+func (s *Swap) verifyHandshake(msg interface{}) error {
+	handshake := msg.(*SwapHandshakeMsg)
+	var empty common.Address
+	if handshake.Beneficiary == empty {
```

Comment:
To me this looks like a weak check. Can handshake be not empty address but instead something invalid?

Reply:
Added type check bool variable inspection.